content/en/docs/tasks/extend-kubernetes/socks5-proxy-access-api.md

```diff
@@ -0,0 +1,97 @@
+
+The following diagram represents what we're going to achieve in this tutorial:
+
+* We've got a `client machine` from where we're going to create requests to talk to the Kubernetes API
+* The Kubernetes server/API is hosted on a `remote server`.
+* We leverage SSH to create a secure SOCKS5 tunnel between the `local` and the `remote server` on top of which the HTTPS traffic between the client and the Kubernetes API will flow.
+
+{{< mermaid >}}
+graph LR;
+
+  subgraph local[Local client machine]
+  client([client])-- local <br> traffic .->  local_ssh[Local SSH <br> SOCKS5 proxy];
+  end
+  local_ssh[SSH <br>SOCKS5 <br> proxy]-- SSH Tunnel -->SSHd
+  
+  subgraph remote[Remote server]
+  SSHd[SSHd <br> server]-- local traffic -->service1;
+  end
+  client([client])-. proxied HTTPs traffic <br> going through the proxy .->service1[Kubernetes API];
+
+  classDef plain fill:#ddd,stroke:#fff,stroke-width:4px,color:#000;
+  classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff;
+  classDef cluster fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+  class ingress,service1,service2,pod1,pod2,pod3,pod4 k8s;
+  class client plain;
+  class cluster cluster;
+{{</ mermaid >}}
+
+## Using ssh to create a SOCKS5 proxy
+
+This command starts a SOCKS5 proxy between your `client machine` and the `remote server` where the Kubernetes API is listening:
+
+```shell
+ssh -D 8080 -q -N  username@kubernetes_server
+```
+
+* `-D 8080`: opens a SOCKS proxy on local port :8080.
+* `-q`: quiet mode. Causes most warning and diagnostic messages to be suppressed.
+* `-N`: Do not execute a remote command. Useful for just forwarding ports.
+* `username@kubernetes_server`: the remote SSH server where the Kubernetes cluster is running.
+
+## Client configuration
+
+To explore the Kubernetes API you'll first need to instruct your clients to send their queries through the SOCKS5 proxy we created earlier:
+
+for command-line tools we can export the following ENV variable:
+
+```shell
+export HTTPS_PROXY=socks5://localhost:8080
+```
+
+Commands like `cURL` will then route their traffic through the proxy:
+
+{{< note >}}
+In this example localhost will not be the localhost of the `client machine` but the localhost of the `remote server`. It could also be replaced by a hostname that is known by the `remote server` and hosted further in the remote infrastructure.
+{{</ note >}}
+
+```shell
+curl -k -v https://localhost/api
+```
+
+To use the official Kubernetes client `kubectl` with a proxy you'll need to set it in your `~/.kube/config` file:
+
+```yaml
+- cluster:
+      certificate-authority-data: xxx
+      server: https://localhost
+      proxy-url: socks5://localhost:8080
+```
+
+Then you should be able to query the remote cluster through the proxy:
+
+```console
+kubectl get pods
+NAMESPACE     NAME                                     READY   STATUS      RESTARTS   AGE
+kube-system   coredns-85cb69466-klwq8                  1/1     Running     0          5m46s
```

Comment:
We should [separate commands from output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output)!

```suggestion
```shell
kubectl get pods
```shell

```console
NAMESPACE     NAME                                     READY   STATUS      RESTARTS   AGE
kube-system   coredns-85cb69466-klwq8                  1/1     Running     0          5m46s
```

Reply:
make sense, corrected!